go/cmd/dolt/commands/sql.go

```diff
@@ -679,6 +679,20 @@ func runShell(ctx *sql.Context, se *sqlEngine, mrEnv env.MultiRepoEnv) error {
 		currPrompt := fmt.Sprintf("%s> ", ctx.GetCurrentDatabase())
 		shell.SetPrompt(currPrompt)
 		shell.SetMultiPrompt(fmt.Sprintf(fmt.Sprintf("%%%ds", len(currPrompt)), "-> "))
+		roots, err := se.getRoots(ctx)
```

Comment:
If you're going to do this, then extract a method that does it and make all the methods that return a root (execShell, execBatch, etc.) do it there, rather than in the main method.